TTS/TikTok.py

```diff
@@ -55,44 +62,100 @@
-        # print(r.text)
-        vstr = [r.json()["data"]["v_str"]][0]
-        b64d = base64.b64decode(vstr)
+        headers = {
+            "User-Agent": "com.zhiliaoapp.musically/2022600030 (Linux; U; Android 7.1.2; es_ES; SM-G988N; "
+            "Build/NRD90M;tt-ok/3.12.13.1)",
+            "Cookie": f"sessionid={settings.config['settings']['tts']['tiktok_sessionid']}",
+        }
+
+        self._session = requests.Session()
+        # set the headers to the session, so we don't have to do it for every request
+        self._session.headers = headers
+
+    def run(self, text: str, filepath: str, random_voice: bool = False):
+        if random_voice:
+            voice = self.random_voice()
+        else:
+            # if tiktok_voice is not set in the config file, then use a random voice
+            voice = settings.config["settings"]["tts"].get("tiktok_voice", None)
 
+        # get the audio from the TikTok API
+        data = self.get_voices(voice=voice, text=text)
+
+        # check if there was an error in the request
+        status_code = data["status_code"]
+        if status_code != 0:
+            raise TikTokTTSException(status_code, data["message"])
+
+        # decode data from base64 to binary
+        raw_voices = data["data"]["v_str"]
+        decoded_voices = base64.b64decode(raw_voices)
+
+        # write voices to specified filepath
         with open(filepath, "wb") as out:
-            out.write(b64d)
+            out.write(decoded_voices)
+
+    def get_voices(self, text: str, voice: Optional[str] = None) -> dict:
+        """If voice is not passed, the API will try to use the most fitting voice"""
+        # sanitize text
+        text = text.replace("+", "plus").replace("&", "and").replace("r/", "")
+
+        # prepare url request
+        params = {"req_text": text, "speaker_map_type": 0, "aid": 1233}
+
+        if voice is not None:
+            params["text_speaker"] = voice
+
+        # send request
+        try:
+            response = self._session.post(self.BASE_URL, params=params)
+        except ConnectionError:
+            time.sleep(random.randrange(1, 7))
+            response = self._session.post(self.BASE_URL, params=params)
+
+        return response.json()
+
+    @staticmethod
+    def random_voice():
+        return random.choice(eng_voices)
+
+
+class TikTokTTSException(Exception):
+    def __init__(self, code: int, message: str):
+        self._code = code
+        self._message = message
+
+    def __str__(self) -> str:
+        if self._code == 1:
+            return f"Code: {self._message}, reason: probably the aid value isn't correct, message: {self._message}"
+
+        if self._code == 2:
+            return f"Code: {self._message}, reason: the text is too long, message: {self._message}"
+
+        if self._code == 4:
+            return f"Code: {self._message}, reason: the speaker doesn't exist, message: {self._message}"
```

Comment:
`return f"Code: {self._message}, reason: the speaker doesn't exist, message: {self._message}"`

The error message should be returning the error number `f"Code: {self._code}"`  first and not the error message twice.

Reply:
```suggestion
            return f"Code: {self._code}, reason: the speaker doesn't exist, message: {self._message}"
```